tests/graph/backends/test_implementations.py

```diff
@@ -0,0 +1,257 @@
+"""Tests verifying that the graph protocols are implemented.
+
+Copyright (c) 2023 Pixelgen Technologies AB.
+"""
+
+import igraph as ig
+import networkx as nx
+import pytest
+from pixelator.graph.backends.implementations import (
+    IgraphBasedEdge,
+    IgraphBasedEdgeSequence,
+    IgraphBasedVertex,
+    IgraphBasedVertexClustering,
+    IgraphBasedVertexSequence,
+    NetworkxBasedEdge,
+    NetworkxBasedEdgeSequence,
+    NetworkxBasedVertex,
+    NetworkxBasedVertexClustering,
+    NetworkxBasedVertexSequence,
+)
+
+
+@pytest.fixture
+def ig_graph():
+    yield ig.Graph()
+
+
+@pytest.fixture
+def nx_graph():
+    yield nx.Graph()
+
+
+@pytest.fixture
+def ig_vertex(ig_graph):
+    yield IgraphBasedVertex(ig_graph.add_vertex(my_attr="a"))
+
+
+@pytest.fixture
+def nx_vertex():
+    yield NetworkxBasedVertex(index=0, data={"my_attr": "a"})
+
+
+@pytest.mark.parametrize("vertex", ["ig_vertex", "nx_vertex"])
+class TestVertexClassesImplementVertexProtocol:
+    def test_index(self, vertex, request):
+        vertex_inst = request.getfixturevalue(vertex)
```

Comment:
`request` is a pytest built-in fixture that among other things allows us to pick up a fixture dynamically at runtime, like we do here. Docs about it can be found here: https://docs.pytest.org/en/7.1.x/reference/reference.html#std-fixture-request